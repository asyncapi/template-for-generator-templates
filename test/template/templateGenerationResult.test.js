/**
 * @jest-environment node
 */

const { readFile } = require('fs').promises;
const path = require('path');
const Generator = require('@asyncapi/generator');
const dummySpecUrl = 'https://rawcdn.githack.com/asyncapi/generator/v1.0.1/test/docs/dummy.yml';
//you always want to generate to new directory to make sure test runs in clear environment
const outputDir = path.resolve('test/temp/templateGenerationResult', Math.random().toString(36).substring(7));

describe('templateGenerationResult()', () => {
  jest.setTimeout(100000);

  beforeAll(async() => {
    const generator = new Generator('./', outputDir, { forceWrite: true });
    await generator.generateFromURL(dummySpecUrl);
  });

  it('generated correct index.html with diagram source', async () => {
    const index = await readFile(path.join(outputDir, 'index.html'), 'utf8');
    expect(index).toMatchSnapshot();
  });
});
