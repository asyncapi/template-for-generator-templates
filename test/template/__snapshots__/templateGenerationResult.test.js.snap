// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templateGenerationResult() generated correct index.html with diagram source 1`] = `
"
<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\">
    <title>Dummy example with all spec features included</title>
    <link rel=\\"stylesheet\\" href=\\"https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css\\">
    <link rel=\\"stylesheet\\" href=\\"style.css\\">

  </head>  
  <body>
    <div class=\\"container mx-auto px-4\\">        
      <p> 
        <h1>Dummy example with all spec features included</h1>
    
    <h2>
      Channels that you can subscribe to
    </h2>
    <hr />
    <br />
    <div class=\\"container mx-auto px-8\\">
      <ul class=\\"list-disc\\"><li><strong>dummy/channel/without/parameter</strong></li></ul>
    </div>

    
    <h2>
      Channels that you can publish to
    </h2>
    <hr />
    <br />
    <div class=\\"container mx-auto px-8\\">
      <ul class=\\"list-disc\\"><li><strong>dummy/channel/with/{dummy}/parameter/create</strong></li></ul>
    </div>

    
    <h2>Schemas</h2>
    <hr>
    You have the following schemas named in components section:
    <div class=\\"container mx-auto px-8\\">
      <ul class=\\"list-disc\\"><li><a href=\\"schemas/dummyArrayWithArray.html\\">dummyArrayWithArray</a></li><li><a href=\\"schemas/dummyArrayWithObject.html\\">dummyArrayWithObject</a></li><li><a href=\\"schemas/dummyCreated.html\\">dummyCreated</a></li><li><a href=\\"schemas/dummyInfo.html\\">dummyInfo</a></li><li><a href=\\"schemas/dummyObject.html\\">dummyObject</a></li><li><a href=\\"schemas/dummyRecursiveObject.html\\">dummyRecursiveObject</a></li><li><a href=\\"schemas/sentAt.html\\">sentAt</a></li></ul>
    </div>
    <div class=\\"mermaid\\">
      classDiagram
    class dummy {
              string}
          
    class dummyCreated {
              prop1 integer
    prop2 string
    sentAt string
    dummyArrayWithObject array
    dummyArrayWithArray array
    dummyObject object
    }
              dummyCreated --|> sentAt
    dummyCreated --|> dummyArrayWithObject
    dummyCreated --|> dummyArrayWithArray
    dummyCreated --|> dummyObject

    class sentAt {
              string}
          
    class dummyArrayWithObject {
              array}
              dummyArrayWithObject --|> dummyInfo

    class dummyInfo {
              prop1 string
    sentAt string
    }
              dummyInfo --|> sentAt

    class dummyObject {
              dummyObjectProp1 string
    dummyObjectProp2 object
    }
              dummyObject --|> sentAt
    dummyObject --|> dummyRecursiveObject

    class dummyRecursiveObject {
              dummyRecursiveProp1 [CIRCULAR] object
    dummyRecursiveProp2 string
    }
              dummyRecursiveObject --|> dummyObject


    </div>

        Share your feedback with us on <a href=\\"http://twitter.com/@AsyncAPISpec\\">Twitter</a>.
        Don't forget to visit our website <a href=\\"https://www.asyncapi.com\\">https://www.asyncapi.com</a>.
      </p> 
    </div>  

    <script src=\\"https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.min.js\\"></script>
    <script>
      mermaid.initialize({
        startOnLoad: true,
        maxTextSize: 50000,
      });
    </script>

  </body>

</html>
"
`;
